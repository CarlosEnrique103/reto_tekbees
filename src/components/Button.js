import React, { Component } from "react";
import { Pressable, StyleSheet, Text } from "react-native";
import { COLORS } from "../constants/colors";

class Button extends Component {
  render() {
    const { title = "Button", onPress } = this.props;
    return (
      <Pressable style={styles.root} onPress={onPress}>
        <Text style={styles.text}>{title}</Text>
      </Pressable>
    );
  }
}

export default Button;

const styles = StyleSheet.create({
  root: {
    width: "100%",
    padding: 20,
    borderRadius: 15,
    backgroundColor: COLORS.primaryColor,
  },
  text: {
    fontWeight: "bold",
    fontFamily: "Roboto-Black",
    textAlign: "center",
    fontSize: 18,
    color: COLORS.whiteColor,
  },
});
