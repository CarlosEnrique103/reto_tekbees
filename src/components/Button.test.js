import renderer from "react-test-renderer";
import Button from "./Button";

describe("Button component", () => {
  it("renders correctly with default props", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with custom title", () => {
    const tree = renderer.create(<Button title="Custom Title" />).toJSON();
    expect(tree.children[0].children[0]).toBe("Custom Title");
  });
});
