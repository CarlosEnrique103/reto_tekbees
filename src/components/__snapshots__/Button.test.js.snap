// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly with custom title 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#ff7e21",
      "borderRadius": 15,
      "padding": 20,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#fff",
        "fontFamily": "Roboto-Black",
        "fontSize": 18,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    Custom Title
  </Text>
</View>
`;

exports[`Button component renders correctly with default props 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#ff7e21",
      "borderRadius": 15,
      "padding": 20,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#fff",
        "fontFamily": "Roboto-Black",
        "fontSize": 18,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    Button
  </Text>
</View>
`;
